state/account.go

```diff
@@ -139,8 +141,16 @@ func (st *Account) AccountType() int32 {
 	return st.accountType
 }
 
+func (st *Account) isMaxNonce() bool {
+	// for legacy account, nonce will finally +2. so use +2 as overflow flag
+	return st.nonce+2 < st.nonce
+}
+
 // SetPendingNonce sets the pending nonce
 func (st *Account) SetPendingNonce(nonce uint64) error {
+	if st.isMaxNonce() {
```

Comment:
the logic could be movd under switch

Reply:
if movd under switch. st.nonce will already ++ and maybe overflow already. pre-check maybe better

Reply:
Could you give an example?

Reply:
changed already